Add additional C48_ATM Rocoto Job CTests

TerrenceMcGuinness-NOAA opened the pull request:
# Pull Request: Add C48_ATM CTests for Rocoto Job Validation

## Summary

This PR extends the Global Workflow CTest framework with new atmospheric (C48_ATM) test cases for validating individual Rocoto jobs. The tests leverage Rocoto's new `--dryrun` mode to extract job submission commands without spawning actual workflow daemons, enabling isolated job-level testing in CI/CD pipelines.

## New Test Cases

| Test Name | Job | Description |
|-----------|-----|-------------|
| `C48_ATM-gfs_stage_ic` | `JGFS_ATMOS_STAGE_IC` | Validate initial condition staging from ICSDIR to COMROOT |
| `C48_ATM-gfs_tracker` | `JGFS_ATMOS_CYCLONE_TRACKER` | Tropical cyclone tracking algorithm |
| `C48_ATM-gfs_genesis` | `JGFS_ATMOS_CYCLONE_GENESIS` | Tropical cyclone genesis detection |


### Rocoto Dryrun Integration ([separate PR](https://github.com/TerrenceMcGuinness-NOAA/rocoto/pull/1) in Rocoto)

The execute phase leverages Rocoto's `--dryrun` mode to extract job submission commands:

```bash
# Extract job card without submitting or spawning daemons
rocotoboot --dryrun -d workflow.db -w workflow.xml -c CYCLE -t TASK 2> jobcard

# Job card is wrapped in {{...}}, extract it
sed -n '/^{{/,/^}}/p' < jobcard | sed '1d;$d' > job.sub

# Submit extracted job directly
sbatch job.sub
```

### NOTE: ###
**CTest** depend on the above feature. And this feature's progress depends on a stable well test **Rocoto** update.
Our intent here is to stress test this version of **Rocoto** above in the pipelines so we [add it into the PATH of the pipeline](https://github.com/TerrenceMcGuinness-NOAA/global-workflow/blob/4bc023493326f83ce697941895505073de3446e2/.gitlab-ci.yml#L155) codes here in this PR (This does not slow down the work but in fact accelerates the main blocker to **CTest** roll out which does depend on issues #4245 #4220 for this).


This approach:
- ✅ Avoids spawning persistent Rocoto daemon processes
- ✅ Enables job card extraction for CI/CD pipelines
- ✅ Works with existing workflow XML configurations
- ✅ Preserves all environment variables and job settings

## Files Changed

### New Test Case YAML Files

| File | Purpose |
|------|---------|
| `dev/ctests/cases/C48_ATM-gfs_stage_ic.yaml` | Stage IC inputs and validate output files |
| `dev/ctests/cases/C48_ATM-gfs_genesis.yaml` | Genesis detector inputs (21 GRIB2 files) and vitals outputs |
| `dev/ctests/cases/C48_ATM-gfs_tracker.yaml` | Tracker inputs (21 GRIB2 files) and ATCF track outputs |

### Updated CMakeLists.txt

Added new `AddJJOBTest()` calls for each C48_ATM test case:

```cmake
AddJJOBTest(
  CASE "C48_ATM"
  JOB "gfs_stage_ic"
  TEST_DATE "2021032312"
)

AddJJOBTest(
  CASE "C48_ATM"
  JOB "gfs_tracker"
  TEST_DATE "2021032312"
)

AddJJOBTest(
  CASE "C48_ATM"
  JOB "gfs_genesis"
  TEST_DATE "2021032312"
)
```

### Updated Execute Script

Enhanced `execute.sh.in` to:
- Support custom `ROCOTO_PATH` for dryrun-enabled Rocoto builds
- Extract job cards using `--dryrun` mode
- Monitor SLURM job status with timeout handling

### Platform Configuration Updates

| File | Changes |
|------|---------|
| `dev/ci/platforms/config.hera` | Added `ROCOTO_PATH`, `STAGED_CTESTS` settings |
| `dev/ci/platforms/config.hercules` | Added `ROCOTO_PATH`, `STAGED_CTESTS` settings |
| `dev/ci/platforms/config.orion` | Added `ROCOTO_PATH`, `STAGED_CTESTS` settings |
| `dev/ci/platforms/config.wcoss2` | Added `ROCOTO_PATH`, `STAGED_CTESTS` settings |
| `dev/ci/platforms/config.gaeac6` | Added `ROCOTO_PATH`, `STAGED_CTESTS` settings |
| `dev/ci/platforms/config.ursa` | Added `ROCOTO_PATH`, `STAGED_CTESTS` settings |

### GitLab CI Integration

| File | Changes |
|------|---------|
| `.gitlab-ci.yml` | Added CTest include |
| `dev/ci/gitlab-ci-ctests.yml` | CTest setup and execution templates |
| `dev/ci/gitlab-ci-cases.yml` | CTest case label definitions |

## Usage

### Local Execution

```bash
# Source platform configuration
source $HOMEgfs/ush/detect_machine.sh
source $HOMEgfs/dev/ci/platforms/config.$MACHINE_ID

# Build CTests
cd $HOMEgfs/dev/ctests
mkdir -p build && cd build
cmake -S "$HOMEgfs"

# List available tests
ctest -N

# Run all C48_ATM tests at once
ctest -L C48_ATM -j5 --output-on-failure

# Run specific test
ctest -L C48_ATM-gfs_genesis -VV
```
## YAML Test Definition Format

Test cases use Jinja2 templating for flexible file path generation:

```yaml
{% set cyc = TEST_DATE | strftime('%H') %}
{% set PDY = TEST_DATE | to_YMD %}
{% set SRC_DIR = STAGED_CTESTS + '/COMROOT/' + PSLOT %}
{% set DST_DIR = RUNTESTS + '/COMROOT/' + TEST_NAME %}

input_files:
    mkdir:
        - {{ DST_GRIB_DIR }}
    copy:
        - [{{ SRC_GRIB_DIR }}/file.grib2, {{ DST_GRIB_DIR }}/file.grib2]

output_files:
    cmpfiles:
        - [{{ SRC_OUTPUT }}/result.txt, {{ DST_OUTPUT }}/result.txt]
```

## How has this been tested?

### All 5 jobs passing on Hera

```
Terry.McGuinness (hfe07) build (add_ctests_jobs-rocoto_update_test) $ ctest -L C48_ATM -j5
Test project /scratch3/NCEPDEV/global/Terry.McGuinness/global-workflow_forked/dev/ctests/build
      Start  1: test_C48_ATM-gfs_stage_ic_setup
      Start  5: test_C48_ATM-gfs_fcst_seg0_setup
      Start  9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup
      Start 13: test_C48_ATM-gfs_tracker_setup
      Start 17: test_C48_ATM-gfs_genesis_setup
 1/20 Test #17: test_C48_ATM-gfs_genesis_setup ...................   Passed    4.56 sec
      Start 18: test_C48_ATM-gfs_genesis_stage
 2/20 Test  #9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup ......   Passed    4.57 sec
      Start 10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage
 3/20 Test  #5: test_C48_ATM-gfs_fcst_seg0_setup .................   Passed    4.57 sec
      Start  6: test_C48_ATM-gfs_fcst_seg0_stage
 4/20 Test  #1: test_C48_ATM-gfs_stage_ic_setup ..................   Passed    4.58 sec
      Start  2: test_C48_ATM-gfs_stage_ic_stage
 5/20 Test #13: test_C48_ATM-gfs_tracker_setup ...................   Passed    4.58 sec
      Start 14: test_C48_ATM-gfs_tracker_stage
 6/20 Test  #2: test_C48_ATM-gfs_stage_ic_stage ..................   Passed    2.63 sec
      Start  3: test_C48_ATM-gfs_stage_ic_execute
 7/20 Test #10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage ......   Passed    2.77 sec
      Start 11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute
 8/20 Test  #6: test_C48_ATM-gfs_fcst_seg0_stage .................   Passed    4.00 sec
      Start  7: test_C48_ATM-gfs_fcst_seg0_execute
 9/20 Test #18: test_C48_ATM-gfs_genesis_stage ...................   Passed   10.40 sec
      Start 19: test_C48_ATM-gfs_genesis_execute
10/20 Test #14: test_C48_ATM-gfs_tracker_stage ...................   Passed   10.38 sec
      Start 15: test_C48_ATM-gfs_tracker_execute
11/20 Test  #3: test_C48_ATM-gfs_stage_ic_execute ................   Passed   62.11 sec
      Start  4: test_C48_ATM-gfs_stage_ic_validate
12/20 Test  #4: test_C48_ATM-gfs_stage_ic_validate ...............   Passed    1.41 sec
13/20 Test #15: test_C48_ATM-gfs_tracker_execute .................   Passed  121.53 sec
      Start 16: test_C48_ATM-gfs_tracker_validate
14/20 Test #16: test_C48_ATM-gfs_tracker_validate ................   Passed    1.35 sec
15/20 Test #11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute ....   Passed  302.06 sec
      Start 12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate
16/20 Test #12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate ...   Passed    2.61 sec
17/20 Test #19: test_C48_ATM-gfs_genesis_execute .................   Passed  541.81 sec
      Start 20: test_C48_ATM-gfs_genesis_validate
18/20 Test #20: test_C48_ATM-gfs_genesis_validate ................   Passed    1.31 sec
19/20 Test  #7: test_C48_ATM-gfs_fcst_seg0_execute ...............   Passed  1562.10 sec
      Start  8: test_C48_ATM-gfs_fcst_seg0_validate
20/20 Test  #8: test_C48_ATM-gfs_fcst_seg0_validate ..............   Passed    1.68 sec

100% tests passed, 0 tests failed out of 20

Label Time Summary:
C48_ATM-gfs_atmos_prod_f000-f002    = 312.01 sec*proc (4 tests)
C48_ATM-gfs_fcst_seg0               = 1572.35 sec*proc (4 tests)
C48_ATM-gfs_genesis                 = 558.07 sec*proc (4 tests)
C48_ATM-gfs_stage_ic                =  70.73 sec*proc (4 tests)
C48_ATM-gfs_tracker                 = 137.84 sec*proc (4 tests)

Total Test time (real) = 1572.37 sec
Terry.McGuinness (hfe07) build (add_ctests_jobs-rocoto_update_test) $ 
```

All 5 test passing on Gaea C6:

```
Terry.McGuinness (gaea68) build (add_C48_ATM_CTests) $ ctest -L C48_ATM -j20
Test project /gpfs/f6/drsa-precip3/scratch/Terry.McGuinness/global-workflow_forked/dev/ctests/build
      Start  1: test_C48_ATM-gfs_stage_ic_setup
      Start  5: test_C48_ATM-gfs_fcst_seg0_setup
      Start  9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup
      Start 13: test_C48_ATM-gfs_tracker_setup
      Start 17: test_C48_ATM-gfs_genesis_setup
 1/20 Test #17: test_C48_ATM-gfs_genesis_setup ...................   Passed    3.06 sec
      Start 18: test_C48_ATM-gfs_genesis_stage
 2/20 Test  #1: test_C48_ATM-gfs_stage_ic_setup ..................   Passed    3.07 sec
      Start  2: test_C48_ATM-gfs_stage_ic_stage
 3/20 Test  #9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup ......   Passed    3.07 sec
      Start 10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage
 4/20 Test #13: test_C48_ATM-gfs_tracker_setup ...................   Passed    3.07 sec
      Start 14: test_C48_ATM-gfs_tracker_stage
 5/20 Test  #5: test_C48_ATM-gfs_fcst_seg0_setup .................   Passed    3.07 sec
      Start  6: test_C48_ATM-gfs_fcst_seg0_stage
 6/20 Test #10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage ......   Passed    1.70 sec
      Start 11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute
 7/20 Test  #2: test_C48_ATM-gfs_stage_ic_stage ..................   Passed    1.71 sec
      Start  3: test_C48_ATM-gfs_stage_ic_execute
 8/20 Test  #6: test_C48_ATM-gfs_fcst_seg0_stage .................   Passed    1.88 sec
      Start  7: test_C48_ATM-gfs_fcst_seg0_execute
 9/20 Test #14: test_C48_ATM-gfs_tracker_stage ...................   Passed  685.04 sec
      Start 15: test_C48_ATM-gfs_tracker_execute
10/20 Test #18: test_C48_ATM-gfs_genesis_stage ...................   Passed  685.05 sec
      Start 19: test_C48_ATM-gfs_genesis_execute
11/20 Test  #3: test_C48_ATM-gfs_stage_ic_execute ................   Passed  844.23 sec
      Start  4: test_C48_ATM-gfs_stage_ic_validate
12/20 Test  #4: test_C48_ATM-gfs_stage_ic_validate ...............   Passed    1.80 sec
13/20 Test #15: test_C48_ATM-gfs_tracker_execute .................   Passed  213.12 sec
      Start 16: test_C48_ATM-gfs_tracker_validate
14/20 Test #16: test_C48_ATM-gfs_tracker_validate ................   Passed    1.71 sec
15/20 Test #11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute ....   Passed  5533.46 sec
      Start 12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate
16/20 Test #12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate ...   Passed    1.79 sec
17/20 Test #19: test_C48_ATM-gfs_genesis_execute .................   Passed  5025.02 sec
      Start 20: test_C48_ATM-gfs_genesis_validate
18/20 Test #20: test_C48_ATM-gfs_genesis_validate ................   Passed    1.70 sec
19/20 Test  #7: test_C48_ATM-gfs_fcst_seg0_execute ...............   Passed  6134.40 sec
      Start  8: test_C48_ATM-gfs_fcst_seg0_validate
20/20 Test  #8: test_C48_ATM-gfs_fcst_seg0_validate ..............   Passed    2.31 sec

100% tests passed, 0 tests failed out of 20

Label Time Summary:
C48_ATM-gfs_atmos_prod_f000-f002    = 5540.02 sec*proc (4 tests)
C48_ATM-gfs_fcst_seg0               = 6141.66 sec*proc (4 tests)
C48_ATM-gfs_genesis                 = 5714.83 sec*proc (4 tests)
C48_ATM-gfs_stage_ic                = 850.81 sec*proc (4 tests)
C48_ATM-gfs_tracker                 = 902.94 sec*proc (4 tests)

Total Test time (real) = 6141.70 sec
```

And now passes on all Role accounts on: Ursa, Orion. Hercules, Gaea C6 as well:

```
 [role-global@hercules-login-1 build]$ ctest -L C48_ATM -j8
Test project /work2/noaa/global/role-global/GFS_CI_CD/HERCULES/global-workflow_forked/dev/ctests/build
      Start 13: test_C48_ATM-gfs_tracker_setup
      Start  1: test_C48_ATM-gfs_stage_ic_setup
      Start  5: test_C48_ATM-gfs_fcst_seg0_setup
      Start  9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup
      Start 17: test_C48_ATM-gfs_genesis_setup
 1/20 Test  #1: test_C48_ATM-gfs_stage_ic_setup ..................   Passed    1.77 sec
      Start  2: test_C48_ATM-gfs_stage_ic_stage
 2/20 Test #13: test_C48_ATM-gfs_tracker_setup ...................   Passed    1.77 sec
      Start 14: test_C48_ATM-gfs_tracker_stage
 3/20 Test  #9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup ......   Passed    1.79 sec
      Start 10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage
 4/20 Test  #5: test_C48_ATM-gfs_fcst_seg0_setup .................   Passed    1.80 sec
      Start  6: test_C48_ATM-gfs_fcst_seg0_stage
 5/20 Test #17: test_C48_ATM-gfs_genesis_setup ...................   Passed    2.14 sec
      Start 18: test_C48_ATM-gfs_genesis_stage
 6/20 Test  #2: test_C48_ATM-gfs_stage_ic_stage ..................   Passed    0.96 sec
      Start  3: test_C48_ATM-gfs_stage_ic_execute
 7/20 Test #10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage ......   Passed    1.22 sec
      Start 11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute
 8/20 Test  #6: test_C48_ATM-gfs_fcst_seg0_stage .................   Passed    1.64 sec
      Start  7: test_C48_ATM-gfs_fcst_seg0_execute
 9/20 Test #14: test_C48_ATM-gfs_tracker_stage ...................   Passed    6.05 sec
      Start 15: test_C48_ATM-gfs_tracker_execute
10/20 Test #18: test_C48_ATM-gfs_genesis_stage ...................   Passed    6.07 sec
      Start 19: test_C48_ATM-gfs_genesis_execute
11/20 Test  #3: test_C48_ATM-gfs_stage_ic_execute ................   Passed   60.62 sec
      Start  4: test_C48_ATM-gfs_stage_ic_validate
12/20 Test  #4: test_C48_ATM-gfs_stage_ic_validate ...............   Passed    1.09 sec
13/20 Test #15: test_C48_ATM-gfs_tracker_execute .................   Passed   60.48 sec
      Start 16: test_C48_ATM-gfs_tracker_validate
14/20 Test #16: test_C48_ATM-gfs_tracker_validate ................   Passed    0.97 sec
15/20 Test #11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute ....   Passed  180.48 sec
      Start 12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate
16/20 Test #12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate ...   Passed    1.35 sec
17/20 Test #19: test_C48_ATM-gfs_genesis_execute .................   Passed  420.80 sec
      Start 20: test_C48_ATM-gfs_genesis_validate
18/20 Test #20: test_C48_ATM-gfs_genesis_validate ................   Passed    1.19 sec
19/20 Test  #7: test_C48_ATM-gfs_fcst_seg0_execute ...............   Passed  1140.74 sec
      Start  8: test_C48_ATM-gfs_fcst_seg0_validate
20/20 Test  #8: test_C48_ATM-gfs_fcst_seg0_validate ..............   Passed    3.91 sec

100% tests passed, 0 tests failed out of 20

Label Time Summary:
C48_ATM-gfs_atmos_prod_f000-f002    = 184.85 sec*proc (4 tests)
C48_ATM-gfs_fcst_seg0               = 1148.09 sec*proc (4 tests)
C48_ATM-gfs_genesis                 = 430.19 sec*proc (4 tests)
C48_ATM-gfs_stage_ic                =  64.44 sec*proc (4 tests)
C48_ATM-gfs_tracker                 =  69.27 sec*proc (4 tests)

Total Test time (real) = 1148.12 sec
```

## Dependencies

### Rocoto Dryrun Feature

This PR depends on Rocoto's `--dryrun` feature which is currently in development:
- **Branch**: `dry_run_no_daemon` 
- **Repository**: [TerrenceMcGuinness-NOAA/rocoto](https://github.com/TerrenceMcGuinness-NOAA/rocoto)
- **Status**: PR pending upstream merge

## Checklist

- [x] New YAML test definitions follow framework conventions
- [x] CMakeLists.txt updated with new test cases
- [x] Execute script uses Rocoto `--dryrun` mode
- [x] Platform configurations updated with required paths
- [x] GitLab CI templates support CTest execution
- [x] README documentation updated
- [x] Tested on Hera with C48_ATM cases
- [ ] Rocoto dryrun PR merged upstream (blocking for production use)

---

**Author**: Terry McGuinness ([@TerrenceMcGuinness-NOAA](https://github.com/TerrenceMcGuinness-NOAA))  
**Target Branch**: `develop`  
**Source Branch**: `add_C48_ATM_CTests`  
**Related PR**: [Rocoto Dryrun Feature](https://github.com/TerrenceMcGuinness-NOAA/rocoto/pull/1)

TerrenceMcGuinness-NOAA committed:
Forcing the GitLab pipeline to use the Rocoto dryrun build over the systems module load for testing purposes

TerrenceMcGuinness-NOAA committed:
added UTIL path on ci config for platform specifics to custom rocoto build

TerrenceMcGuinness-NOAA committed:
updated path to Rocoto used in pipeline

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into test_rocotoboot_dryrun

TerrenceMcGuinness-NOAA committed:
added paths to rocotoboot dryrun uddate for Ursa in ci platforms config file

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into test_rocotoboot_dryrun

TerrenceMcGuinness-NOAA committed:
fist pass of updated ctest yaml files with sourc dest as keys for readablity

TerrenceMcGuinness-NOAA committed:
Restore C48_ATM CTests to legacy schema

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctest_jobs

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctest_jobs

TerrenceMcGuinness-NOAA committed:
C48_ATM-gfs_stage_ic.yaml now passes

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctest_jobs

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctest_jobs

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into test_rocotoboot_dryrun

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into test_rocotoboot_dryrun

TerrenceMcGuinness-NOAA committed:
needed to update PATH to pickup the rocotoboot we are testing

TerrenceMcGuinness-NOAA committed:
forgot to get MACHINE_ID to soruce the correct platform for getting the runtime rocotorun

TerrenceMcGuinness-NOAA committed:
needed to drop the GW on HOMEgfs in execure ctest template

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctest_jobs

TerrenceMcGuinness-NOAA committed:
Merge branch 'test_rocotoboot_dryrun' into add_ctests_jobs-rocoto_update_test

TerrenceMcGuinness-NOAA committed:
update outpuput files for C48_ATM genesis and tracker

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctests_jobs-rocoto_update_test

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctests_jobs-rocoto_update_test

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctests_jobs-rocoto_update_test

TerrenceMcGuinness-NOAA committed:
working Hera runs agains rocoto dryrun updates in T.McG user account

TerrenceMcGuinness-NOAA committed:
Fix C48_ATM-gfs_tracker CTest input staging

The gfs_tracker job expects GRIB2 input files in products/atmos/grib2/0p25, but the test case was previously staging NetCDF history files to model/atmos/history. This mismatch would cause the data_check_gfs.sh script to fail.

This commit updates the test case to stage the correct gfs.t{cyc}z.pgrb2.0p25.f* files to the expected directory, ensuring the test environment matches the job's requirements.

TerrenceMcGuinness-NOAA committed:
Fix C48_ATM-gfs_genesis CTest input staging

The gfs_genesis job expects GRIB2 input files in products/atmos/grib2/0p25, but the test case was previously staging NetCDF history files to model/atmos/history.

Updated to stage gfs.t{cyc}z.pgrb2.0p25.f* files to the correct directory, matching the job's COM_ATMOS_GRIB_GRID_TMPL expectations.

TerrenceMcGuinness-NOAA committed:
got C48_ATM-gfs_tracker working

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_ctests_jobs-rocoto_update_test

TerrenceMcGuinness-NOAA committed:
updated all C48_ATM Rocoto Job CTests in development working and changed ROCOTO_TEST_PATH to ROCOTO_PATH

TerrenceMcGuinness-NOAA committed:
updated where dryrun rocotoboot is ran in T.McG account (needs to be update to role account with PR is ready)

TerrenceMcGuinness-NOAA committed:
updated comments for use of ROCOTO_PATH

TerrenceMcGuinness-NOAA committed:
updated comments for settting and unset-ing the ROCOTO_PATH

TerrenceMcGuinness-NOAA committed:
added working CTests into pipline matrix contol variable

TerrenceMcGuinness-NOAA committed:
new line at enc of config.hear file

TerrenceMcGuinness-NOAA committed:
new line at enc of config.ursa file

TerrenceMcGuinness-NOAA committed:
updated comments regarding ROCOTO_PATH

TerrenceMcGuinness-NOAA committed:
needed CR at end of CMake list file

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
updated path env var to Rocoto with prepend GFS_CI

TerrenceMcGuinness-NOAA committed:
Merge branch 'add_C48_ATM_CTests' of github.com:TerrenceMcGuinness-NOAA/global-workflow into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
added option to run rocotojobcard if it exsists

TerrenceMcGuinness-NOAA committed:
adde rocoto path to hercules config

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
added paths for running custome Rocoto with dryrun feature

TerrenceMcGuinness-NOAA committed:
Merge branch 'add_C48_ATM_CTests' of github.com:TerrenceMcGuinness-NOAA/global-workflow into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
adding custom path to Rocoto with dryrun feature in ci config to point to role account

TerrenceMcGuinness-NOAA committed:
needed bin on the end of full path for Rocoto in CI config

TerrenceMcGuinness-NOAA committed:
adding custom path to Rocoto with dryrun for Orion and Ursa

TerrenceMcGuinness-NOAA committed:
forgot to use GW_ on HOMEgfs in the main pipeline script

TerrenceMcGuinness-NOAA committed:
added issing STAGED_CTESTS in config orion in ci platforms

TerrenceMcGuinness-NOAA committed:
Fix C48_S2SW CTest YAML file naming to match baseline

- gfs_fcst_seg0: Fix typo atm.lo.f003.txt -> log.f003.txt
- gfs_fcst_seg0: Remove gfs.ice./gfs.ocean. prefixes from 41 history files
- ice_prod_f006: Fix input history naming and output paths (native/ subdir)
- ocean_prod_f006: Fix input history naming and output paths (native/ subdir)
- gefs_fcst_mem001_seg0: Remove gefs.ice./gefs.ocean. prefixes from history files

All GFS tests now pass (12/12 phases). GEFS test YAML corrected but
requires longer runtime for full validation.

TerrenceMcGuinness-NOAA committed:
Merge branch 'add_C48_ATM_CTests' of github.com:TerrenceMcGuinness-NOAA/global-workflow into add_C48_ATM_CTests

DavidHuber-NOAA commented:
@TerrenceMcGuinness-NOAA can you check the description of this PR and ensure it is still up to date? There are some mentions to `rocoto --dryrun`, but I also see in the code the use of `rocotojobcard`.  I'm just wondering which is getting used.

Also, can you point me to the `rocoto` branch that this is being developed against?

@DavidHuber-NOAA Yeah I should take  `rocotojobcard` out. It just came across a use case that failed ~T.McG

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Merge branch 'NOAA-EMC:develop' into add_C48_ATM_CTests

DavidHuber-NOAA commented on .gitlab-ci.yml:
```diff
@@ -147,6 +147,17 @@ include:
       echo "Setting up build environment for ${machine^}..."
       echo "Using build directory ${GW_HOMEgfs}"
+
+      # Load custom Rocoto path if specified for user controle of Rocoto version
```
```suggestion
      # Load custom Rocoto path if specified for user control of Rocoto version
```

DavidHuber-NOAA commented on dev/ci/gitlab-ci-cases.yml:
```diff
@@ -30,6 +30,18 @@
+      # Load platform-specific configuration first
+      source "${GW_HOMEgfs}/dev/ci/platforms/config.${machine}"
+      if [[ -n "${GFS_CI_ROCOTO_PATH:-}" && -d "${GFS_CI_ROCOTO_PATH}" ]]; then
+        export PATH="${GFS_CI_ROCOTO_PATH}:$PATH"
```
```suggestion
        export PATH="${GFS_CI_ROCOTO_PATH}:${PATH}"
```

DavidHuber-NOAA commented on .gitlab-ci.yml:
```diff
@@ -147,6 +147,17 @@ include:
+      # This allows testing with different Rocoto builds without changing system Rocoto
+      source "${GW_HOMEgfs}/dev/ci/platforms/config.${MACHINE_ID}"
+      if [[ -n "${GFS_CI_ROCOTO_PATH:-}" && -d "${GFS_CI_ROCOTO_PATH}" ]]; then
+        export PATH="${GFS_CI_ROCOTO_PATH}:$PATH"
```
```suggestion
        export PATH="${GFS_CI_ROCOTO_PATH}:${PATH}"
```

DavidHuber-NOAA commented on dev/ci/gitlab-ci-ctests.yml:
```diff
@@ -30,6 +30,18 @@
+    # Load platform-specific configuration first
+    source "${GW_HOMEgfs}/dev/ci/platforms/config.${machine}"
+    if [[ -n "${GFS_CI_ROCOTO_PATH:-}" && -d "${GFS_CI_ROCOTO_PATH}" ]]; then
+      export PATH="${GFS_CI_ROCOTO_PATH}:$PATH"
```
```suggestion
      export PATH="${GFS_CI_ROCOTO_PATH}:${PATH}"
```

DavidHuber-NOAA commented on dev/ci/gitlab-ci-ctests.yml:
```diff
@@ -52,7 +64,15 @@
     source "${GW_HOMEgfs}/dev/ush/gw_setup.sh"
+    # Platform config loaded by gw_setup.sh, now check for custom Rocoto
+    if [[ -n "${GFS_CI_ROCOTO_PATH:-}" && -d "${GFS_CI_ROCOTO_PATH}" ]]; then
+      export PATH="${GFS_CI_ROCOTO_PATH}:$PATH"
```
```suggestion
      export PATH="${GFS_CI_ROCOTO_PATH}:${PATH}"
```

DavidHuber-NOAA commented on dev/ci/platforms/config.wcoss2:
```diff
@@ -2,6 +2,21 @@
+#########################################################################
+
+# Utilities path for custom builds and role account tools
+# TODO: Populate with actual role account path for this machine
```
Would you like me to install this on WCOSS2?

DavidHuber-NOAA replied:
I realize now that WCOSS2 isn't viable just yet as the scripts assume slurm.  I'm happy to install it if you like, though.

TerrenceMcGuinness-NOAA replied:
We have not stress tested the Rocoto updates on the RDHPCS yet.

DavidHuber-NOAA commented on dev/ctests/scripts/execute.sh.in:
```diff
@@ -8,18 +8,28 @@ TEST_NAME=$("${HOMEgfs}/dev/ci/scripts/utils/ci_utils.sh" get_pslot "@CMAKE_CURR
 idate=$3
 
-rocotoboot_dryrun=$(command -v rocotoboot_dryrun) || true
+# Use standard rocotoboot with --dryrun flag
```
I think this comment is out of place.
```suggestion
```

DavidHuber-NOAA reviewed (commented):
Looks good overall. Just a few minor comments.

TerrenceMcGuinness-NOAA committed:
Update dev/ci/gitlab-ci-ctests.yml

missed that one -- thanks

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

TerrenceMcGuinness-NOAA committed:
Update .gitlab-ci.yml

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

TerrenceMcGuinness-NOAA committed:
Update dev/ci/gitlab-ci-cases.yml

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

TerrenceMcGuinness-NOAA committed:
Update .gitlab-ci.yml

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

TerrenceMcGuinness-NOAA committed:
Update dev/ci/gitlab-ci-ctests.yml

Co-authored-by: David Huber <69919478+DavidHuber-NOAA@users.noreply.github.com>

TerrenceMcGuinness-NOAA committed:
removed references to rocotojobcard in CTest executatale bash template.  CTest exists when path to Rocoto is not specified because it requires the feature branched Rocoto

DavidHuber-NOAA reviewed (dismissed):
Looks good.

DavidHuber-NOAA commented:
@TerrenceMcGuinness-NOAA Can you please rerun the ctests and post results with the run directory?

TerrenceMcGuinness-NOAA committed:
added test for DATAROOT cleanup before running Rocoto ctest job

TerrenceMcGuinness-NOAA committed:
Fix ocean/ice output filenames in C48_S2SWA_gefs CTest YAML

Remove '.ocean.' and '.ice.' prefixes from 24hr_avg history filenames
to match actual model output naming convention:
- gefs.ocean.t12z.24hr_avg.fXXX.nc -> gefs.t12z.24hr_avg.fXXX.nc
- gefs.ice.t12z.24hr_avg.fXXX.nc -> gefs.t12z.24hr_avg.fXXX.nc

Component type is distinguished by directory path (model/ocean/history/
vs model/ice/history/) rather than filename prefix.

TerrenceMcGuinness-NOAA committed:
Merge branch 'add_C48_ATM_CTests' of github.com:TerrenceMcGuinness-NOAA/global-workflow into add_C48_ATM_CTests

TerrenceMcGuinness-NOAA committed:
Fix DATAROOT cleanup in execute.sh.in to suppress config.base debug output

TerrenceMcGuinness-NOAA commented:
Using the [GitLab Scheduled Launches](https://vlab.noaa.gov/gitlab-community/NWS/Operations/NCEP/EMC/global-workflow/-/pipeline_schedules) ran this branch directly from the server on all platforms in advance of having the pipeline changes on the develop branch:

- [Passed all](https://vlab.noaa.gov/gitlab-community/NWS/Operations/NCEP/EMC/global-workflow/-/pipelines/6692) the updated CTests cases in this branch
- [Passed all](https://vlab.noaa.gov/gitlab-community/NWS/Operations/NCEP/EMC/global-workflow/-/pipelines/6690) PR tests against the [**Rocoto** Feature branch](https://github.com/TerrenceMcGuinness-NOAA/rocoto/pull/2) with `--dryrun` and no-daemons

<img width="1091" height="809" alt="image" src="https://github.com/user-attachments/assets/f952cf83-f2d2-4ff6-9fe7-3335cdcdc2e8" />


<img width="1558" height="898" alt="image" src="https://github.com/user-attachments/assets/ae2a7f3e-38da-4446-a3de-5794811d59fa" />

All the run directories for each run can be found in the role accounts on each system respectively:
(i.e. [C48_ATM-gfs_atmos_prod_f000-f002 GaeaC6](https://vlab.noaa.gov/gitlab-community/NWS/Operations/NCEP/EMC/global-workflow/-/jobs/45323))

```
Test project /scratch3/NCEPDEV/global/role.glopara/GFS_CI_CD/URSA/BUILDS/GITLAB/ctests_0_64861d67_6692/global-workflow/dev/ctests/build
    Start  9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup
1/4 Test  #9: test_C48_ATM-gfs_atmos_prod_f000-f002_setup ......   Passed    1.60 sec
    Start 10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage
2/4 Test #10: test_C48_ATM-gfs_atmos_prod_f000-f002_stage ......   Passed    1.22 sec
    Start 11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute
3/4 Test #11: test_C48_ATM-gfs_atmos_prod_f000-f002_execute ....   Passed  180.69 sec
    Start 12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate
4/4 Test #12: test_C48_ATM-gfs_atmos_prod_f000-f002_validate ...   Passed    1.07 sec
100% tests passed, 0 tests failed out of 4
Label Time Summary:
C48_ATM-gfs_atmos_prod_f000-f002    = 184.57 sec*proc (4 tests)
Total Test time (real) = 184.58 sec
=========================================
SUCCESS: All CTests passed
=========================================
```

DavidHuber-NOAA commented:
Thank you @TerrenceMcGuinness-NOAA!! Merging.

DavidHuber-NOAA committed:
Merge branch 'develop' into add_C48_ATM_CTests
